java-backend/src/main/java/org/kframework/backend/java/util/Profiler2.java

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2015-2018 K Team. All Rights Reserved.
+package org.kframework.backend.java.util;
+
+import org.kframework.main.Main;
+
+/**
+ * @author Denis Bogdanas
+ * Created on 23-Jul-18.
+ */
+public class Profiler2 {
```

Comment:
Shouldn't this be set up so that each invocation of a K program only has one Profiler2 instance? If so, you are going to need the @RequestScoped attribute.